### ISSUE TITLE
conan-openssl declares libs it doesn't contain 

### ISSUE THREAD

**Issue by michaelmaguire** [conan-community/community#50]
### Description of Problem, Request, or Question

The conan-openssl package ```package_info()``` for visual studio declares:

```
CONAN_LIBS_OPENSSL[ssleay32MD;libeay32MD;crypt32.lib;msi.lib]
```

but only packages up ```ssleay32MD``` and ```libeay32MD```

### Package Details (Include if Applicable)
* Package Name/Version: **openssl/1.0.2p**
* Operating System: **Windows Server 2012 R2**
* Compiler+version: **Visual Studio 14 2015 v140_xp**
* Docker image: **N/A**

### Steps to reproduce (Include if Applicable)

Preparation: Add a little extra logging to the FindOpenSSL.cmake for OpenSSL:

```
mmaguir1@bnxwnydnxgap29  /E/DevelopBp/mmaguir1/src/nx/conan-openssl
$ git diff
diff --git a/FindOpenSSL.cmake b/FindOpenSSL.cmake
index 695bc55..8a1c626 100644
--- a/FindOpenSSL.cmake
+++ b/FindOpenSSL.cmake
@@ -16,6 +16,10 @@

 MESSAGE(STATUS "********* Conan FindOpenSSL wrapper! **********")

+MESSAGE(STATUS "CONAN_OPENSSL_ROOT[${CONAN_OPENSSL_ROOT}]")
+MESSAGE(STATUS "CONAN_INCLUDE_DIRS_OPENSSL[${CONAN_INCLUDE_DIRS_OPENSSL}]")
+MESSAGE(STATUS "CONAN_LIBS_OPENSSL[${CONAN_LIBS_OPENSSL}]")
+
 SET(OPENSSL_ROOT_DIR ${CONAN_OPENSSL_ROOT})
 SET(OPENSSL_INCLUDE_DIR ${CONAN_INCLUDE_DIRS_OPENSSL})

@@ -36,6 +40,13 @@ FOREACH(THELIB ${CONAN_LIBS_OPENSSL})
 ENDFOREACH()


+MESSAGE(STATUS "OPENSSL_ROOT_DIR[${OPENSSL_ROOT_DIR}]")
+MESSAGE(STATUS "OPENSSL_INCLUDE_DIR[${OPENSSL_INCLUDE_DIR}]")
+MESSAGE(STATUS "OPENSSL_SSL_LIBRARY[${OPENSSL_SSL_LIBRARY}]")
+MESSAGE(STATUS "OPENSSL_CRYPTO_LIBRARY[${OPENSSL_CRYPTO_LIBRARY}]")
+MESSAGE(STATUS "OPENSSL_LIBRARIES[${OPENSSL_LIBRARIES}]")
+
+
 function(from_hex HEX DEC)
   string(TOUPPER "${HEX}" HEX)
   set(_res 0)
```

Now build https://github.com/conan-community/conan-openssl with settings:

```
os=Windows
arch=x86_64
compiler=Visual Studio
compiler.version=14
compiler.toolset=v140_xp
build_type=Release
```

You'll see build output like:

```
openssl/1.0.2p@PORT/stable (test package): Running build()
-- The C compiler identification is MSVC 19.0.24215.1
-- The CXX compiler identification is MSVC 19.0.24215.1
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- ********* Conan FindOpenSSL wrapper! **********
-- CONAN_OPENSSL_ROOT[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- CONAN_INCLUDE_DIRS_OPENSSL[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- CONAN_LIBS_OPENSSL[ssleay32MD;libeay32MD;crypt32.lib;msi.lib]
-- OPENSSL_ROOT_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- OPENSSL_INCLUDE_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- OPENSSL_SSL_LIBRARY[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib]
-- OPENSSL_CRYPTO_LIBRARY[]
-- OPENSSL_LIBRARIES[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;C:/Program Files (x86)/Window
s Kits/8.1/Lib/winv6.3/um/x64/Crypt32.Lib;C:/Program Files (x86)/Windows Kits/8.1/Lib/winv6.3/um/x64/Msi.Lib]
-- Found OpenSSL: C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;C:/Program Files (x86)/Windows K
its/8.1/Lib/winv6.3/um/x64/Crypt32.Lib;C:/Program Files (x86)/Windows Kits/8.1/Lib/winv6.3/um/x64/Msi.Lib (found version "1.0.2p")
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
```

Note especially how it seems to have found  Windows Kits versions of Crypt32.Lib and Msi.Lib:

> -- Found OpenSSL: C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;C:/Program Files (x86)/Windows K
> its/8.1/Lib/winv6.3/um/x64/Crypt32.Lib;C:/Program Files (x86)/Windows Kits/8.1/Lib/winv6.3/um/x64/Msi.Lib (found version "1.0.2p")
> -- Configuring done
> 

Now go and try building a project like https://github.com/bincrafters/conan-libcurl that depends on OpenSSL:

```
-- ********* Conan FindOpenSSL wrapper! **********
-- CONAN_OPENSSL_ROOT[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- CONAN_INCLUDE_DIRS_OPENSSL[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- CONAN_LIBS_OPENSSL[ssleay32MD;libeay32MD;crypt32.lib;msi.lib]
-- OPENSSL_ROOT_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- OPENSSL_INCLUDE_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- OPENSSL_SSL_LIBRARY[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib]
-- OPENSSL_CRYPTO_LIBRARY[]
-- OPENSSL_LIBRARIES[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;LIB_crypt32.lib-NOTFOUND;LIB_
msi.lib-NOTFOUND]
CMake Error at C:/Program Files/CMake/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_LIBRARIES) (found
  version "1.0.2p")
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
```

Questions:

1. Why does this conan-openssl package declare libs (crypt32.lib;msi.lib) that it doesn't actually provide itself?
2. Is this related to a [need to run vcvars64.bat before building OpenSSL](https://github.com/conan-community/community/issues/49) that means I need to also run vcvars64.bat before building any package that depends on OpenSSL?

**Comment by michaelmaguire** [comment]
Additional info: If I try running ```C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64>vcvars64``` first, then it seems to get further, but I see different errors:

```
-- ********* Conan FindOpenSSL wrapper! **********
-- CONAN_OPENSSL_ROOT[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- CONAN_INCLUDE_DIRS_OPENSSL[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- CONAN_LIBS_OPENSSL[ssleay32MD;libeay32MD;crypt32.lib;msi.lib]
-- OPENSSL_ROOT_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d]
-- OPENSSL_INCLUDE_DIR[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/include]
-- OPENSSL_SSL_LIBRARY[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib]
-- OPENSSL_CRYPTO_LIBRARY[]
-- OPENSSL_LIBRARIES[C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;C:/Program Files (x86)/Window
s Kits/8.1/Lib/winv6.3/um/x64/Crypt32.Lib;C:/Program Files (x86)/Windows Kits/8.1/Lib/winv6.3/um/x64/Msi.Lib]
-- Found OpenSSL: C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/ssleay32MD.lib;C:/Users/mmaguir1/.conan/data/openssl/1.0.2p/PORT/stable/package/28e6d62bb9354a736b0f2b12ca7ca07dd9b5909d/lib/libeay32MD.lib;C:/Program Files (x86)/Windows K
its/8.1/Lib/winv6.3/um/x64/Crypt32.Lib;C:/Program Files (x86)/Windows Kits/8.1/Lib/winv6.3/um/x64/Msi.Lib (found version "1.0.2p")
-- Looking for openssl/crypto.h
-- Looking for openssl/crypto.h - found
-- Looking for openssl/err.h
-- Looking for openssl/err.h - found
-- Looking for openssl/pem.h
-- Looking for openssl/pem.h - found
-- Looking for openssl/rsa.h
-- Looking for openssl/rsa.h - found
-- Looking for openssl/ssl.h
-- Looking for openssl/ssl.h - not found
-- Looking for openssl/x509.h
-- Looking for openssl/x509.h - found
-- Looking for openssl/rand.h
-- Looking for openssl/rand.h - found
-- LDAP needs to be enabled to support LDAPS
-- Looking for idn2_lookup_ul in idn2;winmm;ws2_32;advapi32;OpenSSL::SSL;OpenSSL::Crypto
CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (add_executable):
  Target "cmTC_fcec7" links to target "OpenSSL::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at C:/Program Files/CMake/share/cmake-3.10/Modules/CheckLibraryExists.cmake:54 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  CMake/Macros.cmake:9 (check_library_exists)
  CMakeLists_original.txt:494 (check_library_exists_concat)
  CMakeLists.txt:8 (include)


-- Configuring incomplete, errors occurred!
See also "C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeOutput.log".
See also "C:/Users/mmaguir1/.conan/data/libcurl/7.61.1/PORT/stable/build/476691fba8aa676b026399ee5d8674966cc71f18/CMakeFiles/CMakeError.log".
libcurl/7.61.1@PORT/stable:
libcurl/7.61.1@PORT/stable: ERROR: Package '476691fba8aa676b026399ee5d8674966cc71f18' build failed
libcurl/7.61.1@PORT/stable: WARN: Build folder C:\Users\mmaguir1\.conan\data\libcurl\7.61.1\PORT\stable\build\476691fba8aa676b026399ee5d8674966cc71f18
ERROR: libcurl/7.61.1@PORT/stable: Error in build() method, line 150
        self.build_with_cmake()
while calling 'build_with_cmake', line 417
        cmake.configure(source_dir=self.source_subfolder)
        ConanException: Error 1 while executing cd C:\Users\mmaguir1\.conan\data\libcurl\7.61.1\PORT\stable\build\476691fba8aa676b026399ee5d8674966cc71f18 && cmake -G "Visual Studio 14 2015 Win64" -DCONAN_LINK_RUNTIME="/MD" -DCONAN_EXPORTED="1" -DCONAN_COMPILER="Visual Studio" -DCONAN_COMPILER_VERSI
ON="14" -DBUILD_SHARED_LIBS="False" -DCMAKE_INSTALL_PREFIX="C:\Users\mmaguir1\.conan\data\libcurl\7.61.1\PORT\stable\package\476691fba8aa676b026399ee5d8674966cc71f18" -DCONAN_CXX_FLAGS="/MP64" -DCONAN_C_FLAGS="/MP64" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DBUILD_TESTING="False" -DBUILD_CURL_EXE="F
alse" -DCURL_DISABLE_LDAP="True" -DCURL_STATICLIB="True" -DCMAKE_DEBUG_POSTFIX="" -DCMAKE_USE_LIBSSH2="False" -DCMAKE_USE_WINSSL="False" -DCMAKE_USE_OPENSSL="True" -Wno-dev -T "v140_xp" source_subfolder
```

**Comment by SSE4** [comment]
1. `crypt32` and `msi` are import libraries for Windows System DLLs (`crypt32.dll` and `msi.dll`, respectively)
conanfile's `package_info` declares them, as they are required to link with OpenSSL libraries (at least in case of static libraries), as OpenSSL uses corresponding Windows API under the hood and depends on such libraries. I believe it's not the source of the problem, and issue actually has another root cause
2. I don't think it's related to the vcvars issue
I'll take a look carefully into your case and let you know

**Comment by michaelmaguire** [comment]
Thanks for the response.

> conanfile's package_info declares them, as they are required to link with OpenSSL libraries 

So I note we don't usually seem to package_info() declare all libraries we transitively depend on -- I imagine this is usually handled by the Conan package ```requires``` mechanism.  

Can I just confirm my understanding here, that we're declaring crypt32 and msi as a convenience here to help populate consumers' link lines because their necessity is not captured in any Conan package further up the tree?

**Comment by SSE4** [comment]
yes, exactly, it's to help consumers, and we usually do this (otherwise, consumers will fail to link).
although conan handles transitive dependencies for conan packages, system libraries should be explicitly listed in some manner, there are many examples there we're doing this: 
https://github.com/bincrafters/conan-ffmpeg/blob/stable/4.0/conanfile.py#L423
https://github.com/bincrafters/conan-sdl2/blob/stable/2.0.8/conanfile.py#L330
same story for Linux, where several libraries might be required (e.g. `dl`, `rt`, `pthread`, `m`, etc.) and OSX, where we must link with various frameworks (such as `VideoToolbox`, `Cocoa` or `CoreFoundation`).

**Comment by SSE4** [comment]
@michaelmaguire question - is ` FindOpenSSL.cmake` you're modifying standard CMake-supplied file, or are you using custom file?
the issue is that ` FindOpenSSL.cmake` doesn't find Crypto library (`OPENSSL_CRYPTO_LIBRARY` is empty), while conan `OpenSSL` package exports both libraries - `ssleay32MD` and `libeay32MD`

**Comment by michaelmaguire** [comment]
@SSE4 I originally made our internal fork of the recipe from https://github.com/conan-community/conan-openssl/tree/release/1.0.2e, and I am using this FindOpenSSL.cmake unmodified:

https://github.com/conan-community/conan-openssl/blob/release/1.0.2e/FindOpenSSL.cmake

**Comment by SSE4** [comment]
okay, by the way, you have specified version `openssl/1.0.2p` in your initial post, which one is correct, `1.0.2p` or `1.0.2e`?

**Comment by michaelmaguire** [comment]
I originally forked off 1.0.2e, made modifications so it could work with our internal toolchains and Artifactory, and published internally.

Then I tried upgrading to 1.0.2p simply by changing the version in our conanfile, assuming there would be few other changes.

I see now in fact there were some conan changes moving from conan-openssl 1.0.2e to 1.0.2p -- I'm retrying now with a fork based off 1.0.2p.

Sorry if this created extra noise.

**Comment by SSE4** [comment]
at some point we decided to not provide our own `FindOpenSSL.cmake`, as we're expecting it to be working with default one with no issues.
therefore, if you could try to remove `FindOpenSSL.cmake` from your forked recipe and give an another try?

**Comment by michaelmaguire** [comment]
This worked.  I was able to build both conan-libcurl and conan-grpc referencing this conan-openssl package.

To recap, attempting to use branch https://github.com/conan-community/conan-openssl/tree/release/1.0.2e by updating the version to "1.0.2p" had this problem, but https://github.com/conan-community/conan-openssl/commits/release/1.0.2p does not.

( I happened to grap at commit https://bbgithub.dev.bloomberg.com/nx/conan-openssl/commit/321c807d9629a808ef895fd7df3c38e3134eccdc )

Note this branch also didn't need vcvars64.bat to be run first.

Thanks and sorry for the noise.
